python/ray/_private/conda.py

```diff
@@ -52,3 +55,135 @@ def get_conda_bin_executable(executable_name):
         conda_bin_dir = os.path.dirname(os.environ["CONDA_EXE"])
         return os.path.join(conda_bin_dir, executable_name)
     return executable_name
+
+
+def _get_conda_env_name(conda_env_path):
+    conda_env_contents = open(conda_env_path).read() if conda_env_path else ""
+    return "ray-%s" % hashlib.sha1(
+        conda_env_contents.encode("utf-8")).hexdigest()
+
+
+def get_or_create_conda_env(conda_env_path, base_dir=None):
+    """
+    Given a conda YAML, creates a conda environment containing the required
+    dependencies if such a conda environment doesn't already exist. Returns the
+    name of the conda environment, which is based on a hash of the YAML.
+
+    Args:
+        conda_env_path: Path to a conda environment YAML file.
+        base_dir (str, optional): Directory to install the environment into via
```

Comment:
if not specified what does it do?

Reply:
Installs to the default directory (e.g. `/Users/archit/anaconda3/envs/`).  Also in this case, you can activate it just by name instead of specifying the whole path of the env.  I'll add a comment.